actionview/test/template/render_test.rb

```diff
@@ -936,6 +936,17 @@ class CachedCustomer < Customer; end
     assert_equal "Hello: david", ActionView::PartialRenderer.collection_cache.read(key)
   end
 
+  test "template body written to cache with expiration when expires_in set" do
+    customer = Customer.new("jarrett", 2)
+    key = cache_key(customer, "test/_customer")
+    assert_nil ActionView::PartialRenderer.collection_cache.read(key)
+    @view.render(partial: "test/customer", collection: [customer], cached: true, expires_in: 1.hour)
+    assert_equal "Hello: jarrett", ActionView::PartialRenderer.collection_cache.read(key)
+    cache_data = ActionView::PartialRenderer.collection_cache.instance_variable_get("@data")
+    cache_key = cache_data.detect { |k, _v| k.include?("jarrett") }.last
+    assert_predicate(cache_key.expires_at, :present?)
```

Comment:
I would drop the `instance_variable_get`. 
Test should not reach into objects like this when possible and I don't think it's necessary here. Testing that the cache expired after traveling an hour should be enough.
Here's how I would write the test  using the Arrange-Act-Assert pattern:

```suggestion
    customer = Customer.new("jarrett", 2)
    key = cache_key(customer, "test/_customer")
    @view.render(partial: "test/customer", collection: [customer], cached: true, expires_in: 1.hour)
    assert_equal "Hello: jarrett", ActionView::PartialRenderer.collection_cache.read(key)

    travel 2.hours

    assert_nil ActionView::PartialRenderer.collection_cache.read(key)

```

Reply:
Thanks for that suggestion!